### DISCUSSION TITLE
IsSubmitting and isDirty on multiple forms

### DISCUSSION THREAD

**Discussion by ronzyfonzy** [react-hook-form/react-hook-form#6571] [Q&A] [answered]
Firstly a BIG ThankYou @bluebill1049 for this amazing library! 🙇
And now to my (hopefully not to weird) question/example/issue.

So I have multiple forms and a button that triggers SUBMIT on all forms. This is achieved by providing a ref to each form and then triggering the submit event. Here is a Codesandbox example [ MultyFormHook-Base](https://codesandbox.io/s/multyformhook-base-kzkgy) and it looks like this:

<img width="400" alt="CleanShot 2021-09-16 at 21 55 32@2x" src="https://user-images.githubusercontent.com/722464/133677452-066d91e3-aee4-4615-a1c5-3728b43036be.png">

The problem I have is that I want to combine _isDirty_ and _isSubmitting_ form states from all forms and use it on the **Submit All** button. I already have a possible example but it doesn't work as expected. The **Submit All** button doesn't stay in loading state and is also disabled/enabled at wrong states. Here is a Codesandbox example [MultyFormHook-Atempt01](https://codesandbox.io/s/multyformhook-base-kzkgy) and it looks like this: 

![CleanShot 2021-09-16 at 22 05 14](https://user-images.githubusercontent.com/722464/133678656-8893dee6-1887-4b01-94e5-2d1577a1e0d6.gif)

What would be the best approach to handle this?

**Comment by bancek** [comment, marked as answer]
This would be one way to solve it: https://codesandbox.io/s/multyformhook-base-forked-fv7qt?file=/src/App.tsx

**Reply by bluebill1049** [reply]
Thanks @bancek

**Reply by ronzyfonzy** [reply]
`useReducer` 🤦 ... Yes this solves the problem. Thank you @bancek 🙇
